python/lsst/obs/base/gen2to3/translators.py

```diff
@@ -214,17 +215,49 @@ def extract(self, gen2id: dict, skyMap: Optional[BaseSkyMap], skyMapName: Option
 class CalibKeyHandler(KeyHandler):
     """A KeyHandler for master calibration datasets.
     """
+    __slots__ = ("ccdKey",)
 
-    def __init__(self):
+    def __init__(self, ccdKey="ccd"):
+        self.ccdKey = ccdKey
         super().__init__("calibration_label")
 
-    __slots__ = ()
-
     def extract(self, gen2id: dict, skyMap: Optional[BaseSkyMap], skyMapName: Optional[str],
                 datasetTypeName: str) -> Any:
         # Docstring inherited from KeyHandler.extract.
         return makeCalibrationLabel(datasetTypeName, gen2id["calibDate"],
-                                    ccd=gen2id.get("ccd"), filter=gen2id.get("filter"))
+                                    ccd=gen2id.get(self.ccdKey), filter=gen2id.get("filter"))
+
+
+class AbstractFilterKeyHandler(KeyHandler):
+    """KeyHandler for filter keys that should be mapped to AbstractFilters.
+    """
+
+    __slots__ = ("_map",)
+
+    def __init__(self, filterDefinitions):
+        super().__init__("abstract_filter")
+        self._map = {d.physical_filter: d.abstract_filter for d in filterDefinitions
+                     if d.physical_filter is not None}
+
+    def extract(self, gen2id, *args, **kwargs):
+        physical = gen2id["filter"]
+        return self._map.get(physical, physical)
+
+
+class PhysicalFilterKeyHandler(KeyHandler):
+    """KeyHandler for filter keys that should be mapped to AbstractFilters.
+    """
+
+    __slots__ = ("_map",)
+
+    def __init__(self, filterDefinitions):
+        super().__init__("physical_filter")
+        self._map = {d.abstract_filter: d.physical_filter for d in filterDefinitions
+                     if d.abstract_filter is not None}
+
+    def extract(self, gen2id, *args, **kwargs):
+        filter = gen2id["filter"]
```

Comment:
Use `abstract` as variable name to match the `physical` name used in the previous class?